pangea-sdk/service/audit/api.go

```diff
@@ -592,27 +619,69 @@ func (ee *SearchEvent) VerifyConsistencyProof(publishedRoots map[int]Root) {
 }
 
 func (ee *EventEnvelope) VerifySignature() EventVerification {
-	if ee.Signature == nil || ee.PublicKey == nil || strings.HasPrefix(*ee.PublicKey, "-----") {
+	// Both nil, so NotVerified
+	if ee.Signature == nil && ee.PublicKey == nil {
 		return NotVerified
 	}
 
-	b := pu.CanonicalizeStruct(ee.Event)
+	// If just one nil, it's an error so Failed
+	if ee.Signature == nil || ee.PublicKey == nil {
+		return Failed
+	}
+
+	b, err := pu.CanonicalizeStruct(ee.Event)
+	if err != nil {
+		return NotVerified
+	}
 
 	sig, err := base64.StdEncoding.DecodeString(*ee.Signature)
 	if err != nil {
 		return Failed
 	}
 
-	pubKey, err := base64.StdEncoding.DecodeString(*ee.PublicKey)
+	publicKey, err := ee.getPublicKey()
 	if err != nil {
 		return Failed
 	}
 
-	v := signer.NewVerifierFromPubKey(pubKey)
-	if v.Verify(b, sig) {
-		return Success
+	v, err := signer.NewVerifierFromPubKey(publicKey)
+	if v == nil {
+		return Failed
+	}
+
+	if v != nil {
+		ver, err := v.Verify(b, sig)
+		if err != nil {
+			return NotVerified
+		}
+		if ver {
+			return Success
+		} else {
+			return Failed
+		}
+	}
+	return NotVerified
+}
+
+func (ee EventEnvelope) getPublicKey() (string, error) {
+	// Should never enter this case
+	if ee.PublicKey == nil {
+		return "", errors.New("Public key field nil pointer")
+	}
+
+	pkinfo := make(map[string]any)
+	err := json.Unmarshal([]byte(*ee.PublicKey), &pkinfo)
+	if err == nil {
+		val, ok := pkinfo["key"]
+		if ok {
+			ret := val.(string)
```

Comment:
although it should never happen, this cast will panic if failed. Maybe you should check and return an error:
```
if ret, ok := val.(string); !ok {
    return ...
}
```

Reply:
Fixed!